### PR TITLE
front: scale GET on trains with margins

### LINKED ISSUE #6310: Space-Time chart scales on basic run time, not the one with allowances
### What happened?

The charts scales ont the blue curve instead of the green one, corresponding to the allowances.
image

### What did you expect to happen?

It should scale on the curve with allowances (the grren one) so that we can see both curves entirely. This is the case even with multiple trains. We should always scale this graph on the green curves because they are always longer than the blue ones.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create any train with an allowance
2. Auto-scale the Space-Time Chart

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: internal staging

### OSRD version (top right corner `Account` button > `Informations`)

3a2b863

### PR DESCRIPTION
**Pull request by clarani**
closes #6310

### CODE REVIEW

File: front/src/modules/simulationResult/components/SpaceTimeChart/createSpaceTimeChart.ts [hunk 1]
```diff
@@ -2,64 +2,68 @@ import * as d3 from 'd3';
...
   rotate: boolean
 ): Chart {
-  d3select(`#${chartID}`).remove();
+  d3select(`#${SPACE_TIME_CHART_ID}`).remove();
 
-  const xValues: T[] = dataSimulation
+  const resetChart = chart === undefined || reset;
+
+  const timeValues = trains
     .map((train) =>
```
Comment (Akctarus) [inline front/src/modules/simulationResult/components/SpaceTimeChart/createSpaceTimeChart.ts]: Can't we use .flatMap() as we have a .map() and a .flat() ?

```suggestion
   .flatMap((train) =>
    (train.eco_headPosition ?? train.headPosition).flatMap((section) =>
      section.map((position) => position.time)
    )
  ) as Date[];
```

**Review by Akctarus — COMMENTED** [review]
Tested and LGTM, just left 1 comment

File: front/src/modules/simulationResult/components/SpaceTimeChart/createSpaceTimeChart.ts (hunk 1 above)
Comment (clarani) [reply front/src/modules/simulationResult/components/SpaceTimeChart/createSpaceTimeChart.ts]: flatMap is more direct, you can read documentation about it here : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap :blush:
As explained in the doc, flatMap() is slightly more efficient than map().flat()

Comment (clarani) [reply front/src/modules/simulationResult/components/SpaceTimeChart/createSpaceTimeChart.ts]: Thanks, I misunderstood ! Done :white_check_mark:

**Review by Akctarus — APPROVED** [review]

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested

**Pull request merged**